x-pack/metricbeat/module/azure/client_test.go

```diff
@@ -274,49 +274,58 @@ func TestBuildBuildTimespan(t *testing.T) {
 	t.Run("Collection period greater than the time grain (PT1M metric every 5 minutes)", func(t *testing.T) {
 		referenceTime, _ := time.Parse(time.RFC3339, "2024-07-30T18:56:00Z")
 		timeGain := "PT1M"
-		collectionPeriod := 5 * time.Minute
+		cfg := Config{
+			Period: 5 * time.Minute,
+		}
 
-		timespan := buildTimespan(referenceTime, timeGain, collectionPeriod)
+		timespan := buildTimespan(referenceTime, timeGain, cfg)
 
 		assert.Equal(t, "2024-07-30T18:51:00Z/2024-07-30T18:56:00Z", timespan)
 	})
 
 	t.Run("Collection period equal to time grain (PT1M metric every 1 minutes)", func(t *testing.T) {
 		referenceTime, _ := time.Parse(time.RFC3339, "2024-07-30T18:56:00Z")
 		timeGain := "PT1M"
-		collectionPeriod := 1 * time.Minute
+		cfg := Config{
+			Period: 1 * time.Minute,
+		}
 
-		timespan := buildTimespan(referenceTime, timeGain, collectionPeriod)
+		timespan := buildTimespan(referenceTime, timeGain, cfg)
 
 		assert.Equal(t, "2024-07-30T18:55:00Z/2024-07-30T18:56:00Z", timespan)
 	})
 
 	t.Run("Collection period equal to time grain (PT5M metric every 5 minutes)", func(t *testing.T) {
 		referenceTime, _ := time.Parse(time.RFC3339, "2024-07-30T18:56:00Z")
 		timeGain := "PT5M"
-		collectionPeriod := 5 * time.Minute
+		cfg := Config{
+			Period: 5 * time.Minute,
+		}
 
-		timespan := buildTimespan(referenceTime, timeGain, collectionPeriod)
+		timespan := buildTimespan(referenceTime, timeGain, cfg)
```

Comment:
nit:

```suggestion
		timespan := buildTimespan(referenceTime, timeGain, cfg.Period)
```

Reply:
I prefer to pass the whole `config` instead of the individual `period` and `latency` fields to make the function arguments shorter and the function implementation more expressive.